lib/Controller/PageController.php

```diff
@@ -185,6 +187,11 @@ public function authorize(
 			return new RedirectResponse($errorRedirectUri);
 		}
 
+		// trusted clients get their auth code back directly
+		if ($client->getTrusted()) {
+			return $this->generateAuthorizationCode($response_type, $client_id, $redirect_uri, $state, $code_challenge, $code_challenge_method);
```

Comment:
Should we check for the code_challenge not to be null, or is it handled inside?

Reply:
No need to check for null, the code can handle it if it is null.